packages/api/test/name.spec.js

```diff
@@ -38,3 +42,128 @@ describe('POST /name/:key', () => {
     assert.strictEqual(resolved.value, value)
   })
 })
+
+describe('GET /name/:key/watch', () => {
+  it('watches for publishes to a key', async () => {
+    const token = await getTestJWT()
+    const name0 = await createNameKeypair()
+    const name1 = await createNameKeypair()
+    const name0Value0 = '/ipfs/bafybeiauyddeo2axgargy56kwxirquxaxso3nobtjtjvoqu552oqciudrm'
+    const name0Value1 = '/ipfs/bafkreiem4twkqzsq2aj4shbycd4yvoj2cx72vezicletlhi7dijjciqpui'
+    const name1Value0 = '/ipfs/bafkreid7fbwjx4swwewit5txzttoja4t4xnkj3rx3q7dlbj76gvixuq35y'
+    const name1Value1 = '/ipfs/bafybeihiyjghsq7gob7vj3vurqf5i4eth3h57ixpaajdxtvi7p4snhag2a'
+    const name0Record0 = await createNameRecord(name0.privateKey, name0Value0)
+    const name0Record1 = await updateNameRecord(name0.privateKey, name0Record0, name0Value1)
+    const name1Record0 = await createNameRecord(name1.privateKey, name1Value0)
+    const name1Record1 = await updateNameRecord(name1.privateKey, name1Record0, name1Value1)
+
+    const publishRecord = async (key, record) => {
+      const publishRes = await fetch(new URL(`name/${key}`, endpoint), {
+        method: 'POST',
+        headers: { Authorization: `Bearer ${token}` },
+        body: uint8arrays.toString(record, 'base64pad')
+      })
+      assert(publishRes.ok)
+    }
+
+    // listen for updates to name0
+    /** @type {import('websocket').connection} */
+    const conn = await new Promise((resolve, reject) => {
+      const client = new WebSocketClient()
+      client.connect(new URL(`name/${name0.id}/watch`, endpoint).toString())
+      client.on('connect', resolve).on('connectFailed', reject)
+    })
+
+    // we're going to publish on two keys, but we're only listening on one
+    // so we should only receive 2 messages.
+    const expectedMsgCount = 2
+    /** @type {import('websocket').Message[]} */
+    const msgs = []
+    const deferred = defer()
+    conn.on('message', msg => {
+      msgs.push(msg)
+      if (msgs.length >= expectedMsgCount) {
+        deferred.resolve()
+      }
```

Comment:
should we throw if `msgs.length > expectedMsgCount` and change this condition to `===` instead?

Reply:
There's an assertion below that checks `msgs.length` === `expectedMsgCount`. If we throw here we'll get an uncaught exception...